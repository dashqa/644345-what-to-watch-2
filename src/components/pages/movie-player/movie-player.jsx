import React from "react";
import {compose} from "recompose";
import PropTypes from "prop-types";

import SvgButton from "@partials/svg-button/svg-button";
import VideoPlayer from "@partials/video-player/video-player";

import withLoading from "@hocs/with-loading/with-loading";
import withCurrentMovie from "@hocs/with-current-movie/with-current-movie";
import withPlayerFunctionality from "@hocs/with-player-functionality/with-player-functionality";
import VideoProgressBar from "@partials/video-player-progress/video-player-progress";

const MoviePlayer = ({currentMovie, videoRef, onPlayPause, onClosePlayer, onFullScreen, isPlaying}) => {
  const {name, videoLink, previewImage} = currentMovie;

  return (
    <div className="player">
      <VideoPlayer
        videoRef={videoRef}
        classes="player__video"
        link={videoLink}
        poster={previewImage}
        width="100%"
        height="100%"
      />

      <button
        type="button"
        className="player__exit"
        onClick={onClosePlayer}
      >
        Exit</button>

      <div className="player__controls">
        <VideoProgressBar
          videoRef={videoRef}
        />

        <div className="player__controls-row">
          <SvgButton
            classes="player__play"
            title="Play"
            svgXlink={isPlaying ? `#pause` : `#play-s`}
            svgViewBox="0 0 19 19"
            svgWidth="19"
            svgHeight="19"
            onClick={onPlayPause}
          />
          <div className="player__name">{name}</div>

          <SvgButton
            classes="player__full-screen"
            title="Full screen"
            svgXlink="#full-screen"
            svgViewBox="0 0 27 27"
            svgWidth="27"
            svgHeight="27"
            onClick={onFullScreen}
          />
        </div>
      </div>
    </div>
  );
};

MoviePlayer.defaultProps = {
  currentMovie: {
    name: ``,
    videoLink: ``,
    previewImage: ``,
    runTime: 0,
  },
  videoRef: {},
  onPlayPause: () => {},
  onClosePlayer: () => {},
  onFullScreen: () => {},
  isPlaying: false,
};

MoviePlayer.propTypes = {
  currentMovie: PropTypes.shape({
    name: PropTypes.string.isRequired,
    videoLink: PropTypes.string.isRequired,
    previewImage: PropTypes.string.isRequired,
    runTime: PropTypes.number.isRequired,
  }).isRequired,
  videoRef: PropTypes.object.isRequired,
  onPlayPause: PropTypes.func.isRequired,
  onClosePlayer: PropTypes.func.isRequired,
  onFullScreen: PropTypes.func.isRequired,
  isPlaying: PropTypes.bool.isRequired,
};

export {MoviePlayer};

export default compose(
    withLoading,
    withCurrentMovie,
    withPlayerFunctionality
)(MoviePlayer);
