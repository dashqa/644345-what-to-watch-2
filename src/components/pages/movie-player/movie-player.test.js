import React from 'react';
import renderer from 'react-test-renderer';
import {MoviePlayer} from './movie-player';

jest.mock(`@partials/video-player/video-player.jsx`, () => `VideoPlayer`);
jest.mock(`@partials/video-player-progress/video-player-progress.jsx`, () => `VideoPlayerProgress`);

const props = {
  currentMovie: {
    name: `Aviator`,
    videoLink: `http://video.com/video.mp4`,
    previewImage: `http://images.com/image1.png`,
    runTime: 100,
  },
  videoRef: React.createRef(),
  onPlayPause: () => {},
  onClosePlayer: () => {},
  onFullScreen: () => {},
  isPlaying: false,
};

it(`Movie player correctly renders after relaunch`, () => {
  const tree = renderer.create(<MoviePlayer {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
