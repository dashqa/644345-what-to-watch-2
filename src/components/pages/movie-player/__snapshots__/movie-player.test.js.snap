// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie player correctly renders after relaunch 1`] = `
<div
  className="player"
>
  <VideoPlayer
    classes="player__video"
    height="100%"
    link="http://video.com/video.mp4"
    poster="http://images.com/image1.png"
    videoRef={
      Object {
        "current": null,
      }
    }
    width="100%"
  />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <VideoPlayerProgress
      videoRef={
        Object {
          "current": null,
        }
      }
    />
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Aviator
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
